internal/tiltfile/value/value.go

```diff
@@ -112,3 +114,23 @@ func StringSliceToList(slice []string) *starlark.List {
 	}
 	return starlark.NewList(v)
 }
+
+// provides dockerfile-style behavior of:
+// a string gets interpreted as a shell command (like, sh -c 'foo bar $X')
+// an array of strings gets interpreted as a raw argv to exec
+func ValueToCmd(v starlark.Value) (model.Cmd, error) {
+	switch x := v.(type) {
+	case nil:
```

Comment:
nit: should this be an error?

(Background: I hate null with a passion)

Reply:
ya, i think this gets at the problem of doing deserialization outside of unpack, where we can't tell the difference between "the user specified explicit nil" and "the user didn't specify the argument". 

i also wonder if the typed nil problem is going to bite us here

Reply:
I've added a commente explaining why this is here.

FWIW, from the UnpackArgs docs:
```
// Beware: an optional *List, *Dict, Callable, Iterable, or Value variable that is
// not assigned is not a valid Starlark Value, so the caller must
// explicitly handle such cases by interpreting nil as None or some
// computed default.
```

(though ideally that comment would explicitly say that an unspecified optional param is left as `nil`)

I only added this case after unit tests in which optional params were unspecified failed because this value is nil, which gives me some level of confidence that we're ok on typed nils, and if we upgrade starlark and that behavior changes, unit tests will catch it.

Also, probably beside the point of this PR, but IIUC, users cannot specify explicit nil in starlark. As hinted at by the starlark comment above, nil is not a valid starlark value. If they pass, e.g. `serve_cmd=None`, then we'll get a `starlark.None` rather than a nil, which we can distinguish post-deserialization. We've still got this problem if we pass go primitives, e.g. `*int` or `*string` to `UnpackArgs`, but I haven't yet noticed any pain from that.